### DISCUSSION TITLE
Intro to Python Resources!

### DISCUSSION THREAD

**Discussion by sudokita** [ag-informatics/ag-informatics-course#35] [General]
I'll make a running list of Intro to Python resources here:

## Free non-credit Purdue Course
For details & registration information: [https://engineering.purdue.edu/Engr/Academics/Undergraduate/ebec](https://engineering.purdue.edu/Engr/Academics/Undergraduate/ebec)

Snippet the course website:
> The College of Engineering is offering Entry-Level Programming in Python to help you build your skills and your resume. The course is open to anyone with a Purdue account, and no previous coding experience is required.
>   
> This course uses Python to introduce fundamental procedural and object-oriented programming concepts. Amongst the many topics covered are such diverse elements as variables, data structures (lists, tuples, dictionaries, and sets), file handling, decision structures, loops, functions and modules, as well as classes and objects, data attributes, and methods.[1](https://engineering.purdue.edu/Engr/Academics/Undergraduate/ebec/#footnote)
>   
> The course content is split over 13 units with each containing reading assignments, recorded lectures, programming exercises and a quiz. The last unit focuses on a comprehensive multi-week final project. Passing typically requires about 6 hours of work per unit. We cover one unit per week in spring and fall semesters, and 2 units per week in the summer semesters. Help is provided through multiple virtual office hours and multiple in-person sessions every week. Learners who successfully complete the course requirements will earn a certificate issued by the College of Engineering.
>  Familiarity with intermediate algebra and a personal computer on which you can install software are required.

**NOTE: There is a summer offering!** 
> During the fall and spring semester, this course is offered as a free non-credit course to all Purdue students, faculty, and staff. This is a low pressure, no stress version of the course. It will not appear on your transcript or affect your GPA, but learners who successfully complete the course requirements will earn a certificate issued by the College of Engineering.
> 
> The summer offering is open to everyone as a 1 credit hour course at the regular tuition rate per credit hour. Current Purdue students can register for the summer semester through the Scheduling Assistant. Others may take this course in the summer by first applying to Purdue as non-degree student, and then registering through the [Scheduling Assistant](https://timetable.mypurdue.purdue.edu/Timetabling/gwt.jsp?page=sectioning#0). Application details for non-degree seeking students can be found [here](https://www.admissions.purdue.edu/nondegree/index.php).

  
## A 2-week Python Self-Study
See details here: [https://github.com/venomj26/Python4DigitalAg](https://github.com/venomj26/Python4DigitalAg)

- Click on the file you'd like to try out (e.g., Demonstration Workbook 1): https://github.com/venomj26/Python4DigitalAg/blob/master/Demonstration%20Workbook%201.ipynb
- You'll need to know how to use github to download / use this, so you can click on the links in the readme of this resource.

**Comment by sudokita** [comment]
Python Cheatsheets: 

Python fundamentals: https://www.datacamp.com/cheat-sheet/python-for-data-science-a-cheat-sheet-for-beginners
Numpy: https://www.datacamp.com/cheat-sheet/numpy-cheat-sheet-data-analysis-in-python

Pandas: https://pandas.pydata.org/Pandas_Cheat_Sheet.pdf
Matplotlib: https://www.datacamp.com/cheat-sheet/matplotlib-cheat-sheet-plotting-in-python

**Comment by tame0001** [comment]
Python Tutorial from Microsoft
YouTube Playlist (3-7 minutes per video): [https://www.youtube.com/watch?v=jFCNu1-Xdsw&list=PLlrxD0HtieHhS8VzuMCfQD4uJ9yne1mE6](https://www.youtube.com/watch?v=jFCNu1-Xdsw&list=PLlrxD0HtieHhS8VzuMCfQD4uJ9yne1mE6)
Learning course (10 modules): [https://learn.microsoft.com/en-us/training/paths/beginner-python/](https://learn.microsoft.com/en-us/training/paths/beginner-python/)
